R/funnelplot.R

```diff
@@ -0,0 +1,774 @@
+#
+# Copyright (C) 2013-2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+FunnelPlot <- function(jaspResults, dataset = NULL, options, ...) {
+
+  if (.fpReady(options)) {
+    dataset <- .fpCheckDataset(jaspResults, dataset, options)
+    .fpH1Fits(jaspResults, dataset, options)
+  }
+
+  # make the funnel plots
+  .fpPlot(jaspResults, dataset, options)
+  if (options[["funnelUnderH1EstimatesTable"]])
+    .fpPlotEstimatesTable(jaspResults, dataset, options)
+
+  # add the funnel plot asymmetry table
+  if (options[["funnelPlotAsymmetryTests"]])
+    .fpTestFunnelPlotAsymmetryTests(jaspResults, dataset, options)
+
+  return()
+}
+
+.fpDependencies <- c("effectSize", "effectSizeStandardError", "split")
+.fpReady        <- function(options) {
+  return(options[["effectSize"]] != "" && options[["effectSizeStandardError"]] != "")
+}
+.fpCheckDataset <- function(jaspResults, dataset, options) {
+
+  # omit NAs
+  dataset <- na.omit(dataset)
+
+  # add a warning message
+  if (!is.null(attr(dataset, "na.action")) && is.null(jaspResults[["missingDataInformation"]])) {
+    missingDataInformation <- createJaspHtml(gettext("Missing Data Summary"))
+    missingDataInformation$position <- 0.1
+    missingDataInformation$dependOn(c(.fpDependencies, "estimatesMappingColor", "estimatesMappingShape", "studyLabel"))
+    missingDataInformation$text <- gettextf("The dataset contains missing values: %1$i missing values were removed from the analysis.", length(attr(dataset, "na.action")))
+    jaspResults[["missingDataInformation"]] <- missingDataInformation
+  }
+
+  .hasErrors(
+    dataset              = dataset,
+    type                 = c("infinity", "observations", "variance"),
+    all.target           = c(
+      options[["effectSize"]],
+      options[["effectSizeStandardError"]]
+    ),
+    observations.amount  = "< 2",
+    exitAnalysisIfErrors = TRUE)
+
+  .hasErrors(
+    dataset              = dataset,
+    seCheck.target       = options[["effectSizeStandardError"]],
+    custom               = .maCheckStandardErrors,
+    exitAnalysisIfErrors = TRUE)
+
+  return(dataset)
+}
+
+.fpH1Fits                       <- function(jaspResults, dataset, options) {
+
+  if (!is.null(jaspResults[["fitContainer"]]))
+    return()
+
+  fitContainer <- createJaspState()
```

Comment:
If you are creating a state why not name it state, this is misleading

Reply:
ah, I see, sure..., I always named them like this